src/toxcore/dht_new/packet.rs

```diff
@@ -0,0 +1,761 @@
+
+use nom::{le_u8, le_u16, be_u64, be_u16, rest};
+
+//use std::cmp::{Ord, Ordering};
+//use std::convert::From;
+//use std::fmt::Debug;
+use std::net::{
+    IpAddr,
+    Ipv4Addr,
+    Ipv6Addr,
+    SocketAddr,
+    SocketAddrV4,
+    SocketAddrV6
+};
+//use std::ops::Deref;
+
+use toxcore::dht_new::binary_io::*;
+use toxcore::crypto_core::*;
+//use toxcore::packet_kind::PacketKind;
+
+use toxcore::dht_new::cookie_factory::*;
+
+/** Standard DHT packet that encapsulates in the encrypted payload
+[`DhtPacketT`](./trait.DhtPacketT.html).
+
+Length      | Contents
+----------- | --------
+`1`         | `uint8_t` [`PacketKind`](../packet_kind/enum.PacketKind.html)
+`32`        | Sender DHT Public Key
+`24`        | Random nonce
+variable    | Encrypted payload
+
+`PacketKind` values for `DhtPacket` can be only `<= 4`.
+
+https://zetok.github.io/tox-spec/#dht-packet
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum DhtPacket {
+    /// [`PingReq`](./struct.PingReq.html) structure.
+    PingReq(PingReq),
+    /// [`PingResp`](./struct.PingResp.html) structure.
+    PingResp(PingResp),
+    /// [`GetN`](./struct.GetN.html) structure.
+    GetNodes(GetNodes),
+    /// [`SendN`](./struct.SendN.html) structure.
+    SendNodes(SendNodes),
+    // /// [`CookieReq`](./struct.CookieReq.html) structure.
+    //CookieReq(CookieReq),
+    // /// [`CookieResp`](./struct.CookieResp.html) structure.
+    // CookieResp(CookieResp),
+    // /// [`CryptoHs`](./struct.CryptoHs.html) structure.
+    //CryptoHs(CryptoHs),
+    /// [`CryptoData`](./struct.CryptoData.html) structure.
+    //CryptoData(CryptoData),
+    /// [`DhtReq`](./struct.DhtReq.html) structure.
+    DhtRequest(DhtRequest),
+    // /// [`LanDisc`](./struct.LanDisc.html) structure.
+    //LanDisc(LanDisc),
+    // /// [`OnionReq0`](./struct.OnionReq0.html) structure.
+    // TODO
+    // OnionReq0(OnionReq0),
+    // /// [`OnionReq1`](./struct.OnionReq1.html) structure.
+    // TODO
+    // OnionReq1(OnionReq1),
+    // /// [`OnionReq2`](./struct.OnionReq2.html) structure.
+    // TODO
+    // OnionReq2(OnionReq2),
+    // /// [`AnnReq`](./struct.AnnReq.html) structure.
+    // AnnReq(AnnReq),
+    // /// [`AnnResp`](./struct.AnnResp.html) structure.
+    // AnnResp(AnnResp),
+    // /// [`OnionDataReq`](./struct.OnionDataReq.html) structure.
+    // TODO
+    // OnionDataReq(OnionDataReq),
+    // /// [`OnionDataResp`](./struct.OnionDataResp.html) structure.
+    // TODO
+    // OnionDataResp(OnionDataResp),
+    // /// [`OnionResp3`](./struct.OnionResp3.html) structure.
+    // TODO
+    // OnionResp3(OnionResp3),
+    // /// [`OnionResp2`](./struct.OnionResp2.html) structure.
+    // TODO
+    // OnionResp2(OnionResp2),
+    // /// [`OnionResp1`](./struct.OnionResp1.html) structure.
+    // TODO
+    // OnionResp1(OnionResp1),
+}
+
+// TODO: max dht_packet size?
+/// Minimal size of [`DhtPacket`](./struct.DhtPacket.html) in bytes.
+pub const DHT_PACKET_MIN_SIZE: usize = 1 // packet type, plain
```

Comment:
We don't need this variable since it is not used anywhere in the code. You may copy it later.